### PR TITLE
Add support for rendering Org readmes

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
This is regarding issue #1768.

Sample project for testing capabilities is `orgreadmetest` (https://deno.land/x/orgreadmetest).

### CODE REVIEW

**Comment by GuiltyDolphin** [comment]
Getting there!

https://deno-website2-biejkizm6-denoland.vercel.app/x/functionality@v0.2.3:

image

Still some WIP though, e.g., code blocks:

https://deno-website2-biejkizm6-denoland.vercel.app/x/dentest@v0.1.0:

image

And some wrapping needs fixing:

https://deno-website2-biejkizm6-denoland.vercel.app/x/dspv@v0.1.0:

image

**Comment by GuiltyDolphin** [comment]
@kt3k It looks like the issue with list content rendering weirdly is to do with the parser rather than the rendering itself. There is an open issue (https://github.com/orgapp/orgajs/issues/91), so I wonder whether it's better to wait for a fix in `orgajs` and then update the code here if needed, or if we should just build a workaround in (not super trivial) here.

Either way, I think we can probably integrate this before a fix as it can always be added later.

Still got some bits to work on, but I'll let you know when I think the code is ready for review. Thanks!

**Comment by GuiltyDolphin** [comment]
Inline footnotes are currently not supported by `orga`. I've made a pull request to add this (https://github.com/orgapp/orgajs/pull/96).

**Comment by GuiltyDolphin** [comment]
I think the only thing left to do (that doesn't need changes in orga) is checkboxes - not sure how we want to style these, but we can always just render as text if need be.

**Comment by GuiltyDolphin** [comment]
@kt3k Ready for review!

Latest version can be seen working at https://deno-website2-lq25ldh19-denoland.vercel.app/x/orgreadmetest@v0.1.3

**Comment by lucacasonato** [comment]
Cool that this works, but I have some questions/concerns:

- I ran lighthouse on this and compared to `main` we load about 50KB more JS for a /x pageload now (including ones that do not render `org`, which is most). That is an increase of around 40-50%. Orga alone is [28.9kb minified](https://bundlephobia.com/package/orga@2.4.8).
- The addition of `date-fns`: why does a markup renderer require a date formatter / parser?
- I am not so thrilled that we now have both `marked` and `reorg` as dependencies. They both cover very similar feature sets.

**Comment by kt3k** [comment]
> I ran lighthouse on this and compared to main we load about 50KB more JS for a /x pageload now (including ones that do not render org, which is most). That is an increase of around 40-50%. Orga alone is 28.9kb minified.

`next build` command shows this increases 120kb -> 141kb first load JS for `/x/[...rest]` page. I don't think this affects that much. (Chrome devtool network tab with JS filtering also shows 169kb -> 195kb load, 26kb/15% increase)

But it's true that most pages don't use .org format. Maybe we can use `next/dynamic` for loading `Org` component lazily. What do you think?

**Comment by lucacasonato** [comment]
@kt3k DevTools showed an extra 50KB bundle for me 🤷‍♂️

**Comment by GuiltyDolphin** [comment]
I'll have a look into minimizing the dependencies.

**Comment by GuiltyDolphin** [comment]
@lucacasonato 


>     * I ran lighthouse on this and compared to `main` we load about 50KB more JS for a /x pageload now (including ones that do not render `org`, which is most). That is an increase of around 40-50%. Orga alone is [28.9kb minified](https://bundlephobia.com/package/orga@2.4.8).

I've switched to lazy loading of the `Org` module, which seems to save space on general page loads (`122kb` first load vs `120kb`, now). I only actually need to use `orga` (directly) for the types - `reorg-parse` is used for the actual parsing. I'm looking into whether I can just require the types files from `orga`.

>     * The addition of `date-fns`: why does a markup renderer require a date formatter / parser?

See https://github.com/orgapp/orgajs/issues/83#issuecomment-826401251. But again, if I can lose the `orga` direct dependency, then maybe we won't need this.

>     * I am not so thrilled that we now have both `marked` and `reorg` as dependencies. They both cover very similar feature sets.

Well `orga` uses `unified`, so we could potentially use a markdown parser from there - might lead to less extra dependencies.

**Comment by GuiltyDolphin** [comment]
@lucacasonato @kt3k I think that's as small as I can gct the deps. Set them to load only when needed though, so we only incur the ~50kb when using a project with an org readme.

File: package.json [hunk 1]
```diff
@@ -32,12 +33,10 @@
     "react-dom": "npm:@preact/compat",
     "react-ssr-prepass": "npm:preact-ssr-prepass",
     "react-transition-group": "^4.4.1",
-    "reorg-parse": "^2.3.16",
     "swr": "^0.4.2",
     "tailwindcss": "^1.9.6",
     "tailwindcss-hyphens": "^0.1.0",
     "timeago-react": "^3.0.2",
-    "unified": "^9.2.1",
```
Comment (kt3k) [inline package.json]: 👍

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by kt3k** [comment]
@GuiltyDolphin Thank you for your contribution! Nice work!

**Pull request merged**